jobspec2/hclutil/blockattrs.go

```diff
@@ -138,39 +178,71 @@ func convertToAttribute(b *hcls.Block) *hcls.Attribute {
 
 		items = append(items, hcls.ObjectConsItem{
 			KeyExpr:   key,
-			ValueExpr: attr.Expr,
+			ValueExpr: attrExpr(attr.Expr),
 		})
 	}
 
-	for _, block := range b.Body.Blocks {
+	for _, blocks := range blocksByType(b.Body.Blocks) {
 		keyExpr := &hcls.ScopeTraversalExpr{
 			Traversal: hcl.Traversal{
 				hcl.TraverseRoot{
-					Name:     block.Type,
-					SrcRange: block.TypeRange,
+					Name:     blocks[0].Type,
+					SrcRange: blocks[0].TypeRange,
 				},
 			},
-			SrcRange: block.TypeRange,
+			SrcRange: blocks[0].TypeRange,
 		}
 		key := &hcls.ObjectConsKeyExpr{
 			Wrapped: keyExpr,
 		}
-		valExpr := convertToAttribute(block).Expr
-		items = append(items, hcls.ObjectConsItem{
+		item := hcls.ObjectConsItem{
 			KeyExpr:   key,
-			ValueExpr: valExpr,
-		})
+			ValueExpr: blocksToExpr(blocks),
+		}
+
+		items = append(items, item)
 	}
 
-	attr := &hcls.Attribute{
-		Name:        b.Type,
-		NameRange:   b.TypeRange,
-		EqualsRange: b.OpenBraceRange,
-		SrcRange:    b.Body.SrcRange,
-		Expr: &hcls.ObjectConsExpr{
-			Items: items,
-		},
+	v := &hcls.ObjectConsExpr{
+		Items: items,
+	}
+
+	for i := len(b.Labels) - 1; i >= 0; i-- {
```

Comment:
I'm having a little trouble understanding this loop. Is this supposed to be accumulating values on `v` instead of overwriting them? It doesn't look like there's any other state outside the loop that's changing.

Reply:
This effectively re-writes the following block:

```hcl
myblock "label1" "label2" "label3" {
  a = "b"
}
```

to

```hcl
{"myblock": [{"label1": [{"label"2: [{"label3": [{"a": "b"}]}]}]}]
```

Which was how hclv1 parsed it.

The implementation loops through the labels in reverse order to accumulate the map layers into v.

Reply:
Ah, I get it now!